lib/Driver/DarwinToolChains.cpp

```diff
@@ -588,76 +589,62 @@ toolchains::Darwin::addDeploymentTargetArgs(ArgStringList &Arguments,
     }
 
     // Compute the platform version.
-    unsigned major, minor, micro;
+    llvm::VersionTuple osVersion;
     if (tripleIsMacCatalystEnvironment(triple)) {
-      triple.getiOSVersion(major, minor, micro);
-
-      // Mac Catalyst on arm was introduced with an iOS deployment target of
-      // 14.0; the linker doesn't want to see a deployment target before that.
-      if (major < 14 && triple.isAArch64()) {
-        major = 14;
-        minor = 0;
-        micro = 0;
-      }
-
-      // Mac Catalyst was introduced with an iOS deployment target of 13.1;
-      // the linker doesn't want to see a deployment target before that.
-      if (major < 13) {
-        major = 13;
-        minor = 1;
-        micro = 0;
+      osVersion = triple.getiOSVersion();
+
+      if (osVersion.getMajor() < 14 && triple.isAArch64()) {
+        // Mac Catalyst on arm was introduced with an iOS deployment target of
+        // 14.0; the linker doesn't want to see a deployment target before that.
+        osVersion = llvm::VersionTuple(/*Major=*/14, /*Minor=*/0);
+      } else if (osVersion.getMajor() < 13) {
+        // Mac Catalyst was introduced with an iOS deployment target of 13.1;
+        // the linker doesn't want to see a deployment target before that.
+        osVersion = llvm::VersionTuple(/*Major=*/13, /*Minor=*/1);
       }
     } else {
       switch (getDarwinPlatformKind((triple))) {
       case DarwinPlatformKind::MacOS:
-        triple.getMacOSXVersion(major, minor, micro);
+        triple.getMacOSXVersion(osVersion);
```

Comment:
@DougGregor `getMacOSXVersion` can actually "fail". In that case `osVersion` is currently still set, but it'll return false. Do we want to handle that at all? I couldn't find a single case where we actually handle the error, so maybe it would be better to just modify the API to ignore the failing case?